applications/DEMApplication/custom_elements/beam_particle.cpp

```diff
@@ -0,0 +1,392 @@
+                               data_buffer.mpOtherParticle,
+                               indentation,
+                               false,
+                               i);
+
+                mContinuumConstitutiveLawArray[i]->ComputeParticleRotationalMoments(this,
+                                                                                    neighbour_iterator,
+                                                                                    equiv_young,
+                                                                                    data_buffer.mDistance,
+                                                                                    calculation_area,
+                                                                                    data_buffer.mLocalCoordSystem,
+                                                                                    ElasticLocalRotationalMoment,
+                                                                                    ViscoLocalRotationalMoment,
+                                                                                    equiv_poisson,
+                                                                                    indentation);
+
+                AddUpMomentsAndProject(data_buffer.mLocalCoordSystem, ElasticLocalRotationalMoment, ViscoLocalRotationalMoment);
+            }
+
+            if (r_process_info[CONTACT_MESH_OPTION] == 1 && (i < (int)mContinuumInitialNeighborsSize) && this->Id() < neighbour_iterator_id) {
+                double total_local_elastic_contact_force[3] = {0.0};
+                total_local_elastic_contact_force[0] = LocalElasticContactForce[0] + LocalElasticExtraContactForce[0];
+                total_local_elastic_contact_force[1] = LocalElasticContactForce[1] + LocalElasticExtraContactForce[1];
+                total_local_elastic_contact_force[2] = LocalElasticContactForce[2] + LocalElasticExtraContactForce[2];
+                CalculateOnContinuumContactElements(i,
+                                                    total_local_elastic_contact_force,
+                                                    contact_sigma,
+                                                    contact_tau,
+                                                    failure_criterion_state,
+                                                    acumulated_damage,
+                                                    time_steps);
+            }
+        } // for each neighbor
+
+        KRATOS_CATCH("")
+    } //  ComputeBallToBallContactForce
+
+    void BeamParticle::Move(const double delta_t, const bool rotation_option, const double force_reduction_factor, const int StepFlag) {
+
+        KRATOS_TRY
+
+        GetTranslationalIntegrationScheme().Move(GetGeometry()[0], delta_t, force_reduction_factor, StepFlag);
+        if (rotation_option) {
+            GetRotationalIntegrationScheme().RotateBeam(GetGeometry()[0], delta_t, force_reduction_factor, StepFlag);
```

Comment:
Here you could call `CalculateRotationalMotionOfRigidBodyElementNode `directly, and avoid incresing the API of the scheme,. Let me know if I am mistaken @joaquinirazabal .

Reply:
You are right!